src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java

```diff
@@ -0,0 +1,1281 @@
+ * common cases:
+ * <ol>
+ *   <li> {@link ShutdownOnSuccess ShutdownOnSuccess} captures the result of the first
+ *   subtask to complete successfully. Once captured, it shuts down the task scope to
+ *   interrupt unfinished threads and wakeup the owner. This class is intended for cases
+ *   where the result of any subtask will do ("invoke any") and where there is no need to
+ *   wait for results of other unfinished subtasks. It defines methods to get the first
+ *   result or throw an exception if all subtasks fail.
+ *   <li> {@link ShutdownOnFailure ShutdownOnFailure} captures the exception of the first
+ *   subtask to fail. Once captured, it shuts down the task scope to interrupt unfinished
+ *   threads and wakeup the owner. This class is intended for cases where the results of all
+ *   subtasks are required ("invoke all"); if any subtask fails then the results of other
+ *   unfinished subtasks are no longer needed. If defines methods to throw an exception if
+ *   any of the subtasks fail.
+ * </ol>
+ *
+ * <p> The following are two examples that use the two classes. In both cases, a pair of
+ * subtasks are forked to fetch resources from two URL locations "left" and "right". The
+ * first example creates a ShutdownOnSuccess object to capture the result of the first
+ * subtask to complete successfully, cancelling the other by way of shutting down the task
+ * scope. The main task waits in {@code join} until either subtask completes with a result
+ * or both subtasks fail. It invokes {@link ShutdownOnSuccess#result(Function)
+ * result(Function)} method to get the captured result. If both subtasks fail then this
+ * method throws a {@code WebApplicationException} with the exception from one of the
+ * subtasks as the cause.
+ * {@snippet lang=java :
+ *     try (var scope = new StructuredTaskScope.ShutdownOnSuccess<String>()) {
+ *
+ *         scope.fork(() -> fetch(left));
+ *         scope.fork(() -> fetch(right));
+ *
+ *         scope.join();
+ *
+ *         // @link regex="result(?=\()" target="ShutdownOnSuccess#result" :
+ *         String result = scope.result(e -> new WebApplicationException(e));
+ *
+ *         ...
+ *     }
+ * }
+ * The second example creates a ShutdownOnFailure object to capture the exception of the
+ * first subtask to fail, cancelling the other by way of shutting down the task scope. The
+ * main task waits in {@link #joinUntil(Instant)} until both subtasks complete with a
+ * result, either fails, or a deadline is reached. It invokes {@link
+ * ShutdownOnFailure#throwIfFailed(Function) throwIfFailed(Function)} to throw an exception
+ * if either subtask fails. This method is a no-op if both subtasks complete successfully.
+ * The example uses {@link Supplier#get()} to get the result of each subtask. Using
+ * {@code Supplier} instead of {@code Subtask} is preferred for common cases where the
+ * the object returned by fork is only used to get the result of a subtask that completed
```

Comment:
```suggestion
 * {@code Supplier} instead of {@code Subtask} is preferred for common cases where
 * the object returned by fork is only used to get the result of a subtask that completed
```